src/Agent.Sdk/Util/VssUtil.cs

```diff
@@ -106,5 +121,26 @@ public static VssCredentials GetVssCredential(ServiceEndpoint serviceEndpoint)
 
             return credentials;
         }
+
+        private static bool CheckSupportOfCustomServerCertificateValidation()
+        {
+            using (var handler = new HttpClientHandler())
+            {
+                handler.ServerCertificateCustomValidationCallback = (message, cert, chain, errors) => { return true; };
```

Comment:
Was it failing on setting up on this callback, or just when the request is being sent?

Reply:
It's failing when the request is being sent.